File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcElement/IfcDistributionElement/IfcDistributionControlElement/IfcAlarm/IfcAlarm.vb
Public Class IfcAlarm Inherits IfcDistributionControlElement
    Public PredefinedType As IfcAlarmTypeEnum

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcAlarmTypeEnum.USERDEFINED) OR ((PredefinedType = IfcAlarmTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
    ' CorrectTypeAssigned : (SIZEOF(IsTypedBy) = 0) OR ('IFC4X3_DEV_73740fe4.IFCALARMTYPE' IN TYPEOF(SELF\IfcObject.IsTypedBy[1].RelatingType))
End Class

File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcElement/IfcDistributionElement/IfcDistributionControlElement/IfcAlarm/IfcAlarm.cs
public class IfcAlarm : IfcDistributionControlElement
{
    public IfcAlarmTypeEnum PredefinedType { get; set; }

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcAlarmTypeEnum.USERDEFINED) OR ((PredefinedType = IfcAlarmTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
    // CorrectTypeAssigned : (SIZEOF(IsTypedBy) = 0) OR ('IFC4X3_DEV_73740fe4.IFCALARMTYPE' IN TYPEOF(SELF\IfcObject.IsTypedBy[1].RelatingType))
}
